tidy3d/components/tcad/doping.py

```diff
@@ -20,12 +20,14 @@ def _normal_dim(self):
         """Returns the normal direction if the box is 2D. False otherwise"""
 
         normal_dim = None
+        inf_dims = 0
         for dim in range(3):
             if self.size[dim] == np.inf:
-                if normal_dim is not None:
-                    raise SetupError("Only 3D and 2D boxes are considered for doping.")
+                inf_dims += 1
                 normal_dim = dim
 
+        if normal_dim is not None and inf_dims < 3:
+            raise SetupError("Only 3D and 2D boxes are considered for doping.")
```

Comment:
I am actually very confused by this function now that I look at it.

- With this modification, it returns *the last* infinite dimension as `normal_dim`? So if I have [inf, 0, inf], `normal_dim` will be 2?
- Previously this would error in this case?
- What if I have a box which has finite dimensions?

I think we've discussed this function before but I'm still baffled by it.

Reply:
Right. I'm still confused by the purpose of this / the definitions. Maybe a quick chat?